front_end/src/components/Campaigns/CampaignForm/CampaignForm.js

```diff
@@ -57,55 +54,86 @@ export default class CampaignForm extends Component {
+                  <div className="onoffswitch">
+                    <input type="checkbox" name="paused" ref="paused" className="onoffswitch-checkbox" id="AccountPaused" defaultChecked={data.paused} value="true"/>
+                    <label className="onoffswitch-label" htmlFor="AccountPaused"></label>
+                  </div>
+                </div>)
+                  : <input type="hidden" name="paused" ref="paused" value={false}/>
+                }
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-xs-4">
+                <div className="form-group">
+                  <label htmlFor="CampaignStartDate">Start Date</label>
+                  <input className="form-control" type="text" id="CampaignStartDate" name="start_date" ref="start_date" defaultValue={formatDate(data.start_date, 'YYYY-MM-DD')} data-parsley-dateformat data-parsley-required />
+                </div>
+              </div>
+              <div className="col-xs-4">
+                <div className="form-group">
+                  <label htmlFor="CampaignEndDate">End Date</label>
+                  <input className="form-control" type="text" id="CampaignEndDate" name="end_date" ref="end_date" defaultValue={formatDate(data.end_date, 'YYYY-MM-DD')} data-parsley-dateformat data-parsley-required />
+                </div>
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-xs-8">
+                <div className="form-group">
+                  <label htmlFor="CampaignCountries">Countries</label><br/>
+                  <select className="form-control js-select" style={{width: '100%'}} id="CampaignCountries" name="countries[]" ref="countries" multiple="multiple" defaultValue={data.countries} data-parsley-required>
+                    {countries}
+                  </select>
+                </div>
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-xs-4">
+                <div className="form-group">
+                  <label htmlFor="CampaignChannelId">Channel</label>
+                  <select className="form-control" id="CampaignChannelId" name="channel_id" ref="channel_id" defaultValue={data.channel_id} data-parsley-required >
+                    {channels}
+                  </select>
+                </div>
+              </div>
+            </div>
           </div>
-          <input onClick={(e) => this.handleFormSubmit(e)} type="submit" value="Submit" className="btn btn-primary"/>
-          {(this.props.editMode)
-            ? <Link to={'/campaigns/' + data.id} className="btn btn-default">Cancel</Link>
-            : <Link to={'/accounts/' + data.account_id} className="btn btn-default">Cancel</Link>
-          }
-          {spinner}
+
+          <div onClick={(e) => this.handleFormSubmit(e)} className="form-submit" >Save {spinner}</div>
```

Comment:
Ah, I see. The problem is that inside `handleSubmitForm` you are using `this` and it will be bound to the `div` DOM node instead of the component. That's why I do this: https://github.com/oyiptong/splice/blob/master/ui/src/containers/Authoring.js#L23

 I am puzzled as to how your way works. But if it does, then :+1: